packages/contentful/scripts/assign-users-as-lab-pi.ts

```diff
@@ -0,0 +1,196 @@
+
+    for (const user of users) {
+      if (
+        !user.fields ||
+        !user.fields.oldLabs ||
+        !user.fields.oldLabs['en-US'].length
+      ) {
+        console.log('No labs, skipping');
+        continue;
+      }
+
+      const labMemberships: Entry[] = user.fields.oldLabs?.['en-US'] || [];
+
+      if (!labMemberships.length) {
+        console.log('No labs, skipping - 2');
+        continue;
+      }
+
+      // Fetch labs referenced in `oldLabs`
+      const labs = await environment.getEntries({
+        content_type: 'labMembership',
+        'sys.id[in]': labMemberships
+          .map((labMembership) => labMembership.sys.id)
+          .join(','),
+      });
+
+      if (!labs.items.length) {
+        console.log(`No labs found for user: ${user.fields.email['en-US']}`);
+        continue;
+      }
+
+      const labIds = labs.items
+        .map((lab) => lab.fields.lab?.['en-US'].sys.id)
+        .filter(Boolean);
+
+      const isPublished = user.isPublished();
+
+      const userEntry = await environment.getEntry(user.sys.id);
+
+      // Update the labs field
+      userEntry.fields.labs = userEntry.fields.labs || {};
+      userEntry.fields.labs['en-US'] = labIds.map((labId) => ({
+        sys: {
+          type: 'Link',
+          linkType: 'Entry',
+          id: labId,
+        },
+      })) as Link<'Entry'>[];
+
+      // Save and publish the updated entry (if it was previously published)
+      try {
+        const updatedUser = await userEntry.update();
+        if (isPublished) {
+          await updatedUser.publish();
+        }
+        console.log(`Updated user ${user.sys.id}`);
+      } catch (err) {
+        console.log(`Error updating user ${user.sys.id}: ${err}`);
+      }
+    }
+
+    console.log('Labs membership rollback completed successfully!');
+  } catch (error) {
+    console.error('Error rolling back Labs membership:', error);
+  }
+}
+
+async function assignLabPI() {
+  try {
+    const space = await client.getSpace(spaceId);
+    const environment: Environment = await space.getEnvironment(environmentId);
+
+    // Fetch all labs
+    const labs = await environment.getEntries({
+      content_type: 'labMembership',
+    });
+
+    for (const lab of labs.items) {
+      if (!lab.fields.lab || !lab.fields.lab['en-US']) {
+        console.log('No lab found, skipping');
+        continue;
+      }
+
+      const labId = lab.fields.lab['en-US'].sys.id;
+
+      // Fetch users who reference this lab in `oldLabs`
+      const users = await environment.getEntries({
+        content_type: 'users',
+        'fields.oldLabs.sys.id': lab.sys.id,
+      } as QueryOptions);
+
+      if (!users.items.length) {
+        console.log(`No users found for lab: ${labId}`);
+        continue;
+      }
+
+      const firstUser = users.items[0]; // Pick the first user - maybe the last user would be better? - also maybe checking if the user is published would be a good idea?
```

Comment:
I believe we should check the role in here. But I'll ask in the engineers channel as well

Reply:
as per the conversation on slack, we should check if the lab memberships has role "Lead PI"  before assigning the user to the `labEntry.fields.labPI`